Source/WebInspectorUI/UserInterface/Models/DOMNodeStyles.js

```diff
@@ -712,10 +743,24 @@ WI.DOMNodeStyles = class DOMNodeStyles extends WI.Object
         // COMPATIBILITY (iOS 13): CSS.CSSRule.groupings did not exist yet.
         let groupings = (payload.groupings || payload.media || []).map((grouping) => {
             let groupingType = WI.CSSManager.protocolGroupingTypeToEnum(grouping.type || grouping.source);
-            let groupingSourceCodeLocation = DOMNodeStyles.createSourceCodeLocation(grouping.sourceURL);
+
+            let location = {};
+            if (payload.sourceRange) {
+                location.line = payload.sourceRange.startLine;
+                location.column = payload.sourceRange.startColumn;
+                location.documentNode = this._node.ownerDocument;
+            }
+
+            let groupingSourceCodeLocation = WI.DOMNodeStyles.createSourceCodeLocation(grouping.sourceURL, location);
             if (styleSheet)
                 groupingSourceCodeLocation = styleSheet.offsetSourceCodeLocation(groupingSourceCodeLocation);
```

Comment:
ditto ([Source/WebInspectorUI/UserInterface/Models/CSSGrouping.js](#diff-09c45dc733ff4be6e27d1fe184904524721db668ad98087bbb46e61b2bb5f267):129)

Reply:
id use a ternary for this
```
let groupingSourceCodeLocation = WI.DOMNodeStyles.createSourceCodeLocation(grouping.sourceURL, location);
let offsetGroupingSourceCodeLocation = styleSheet ? styleSheet.offsetSourceCodeLocation(groupingSourceCodeLocation) : groupingSourceCodeLocation;
```

or even just an optional chaining with nullish coalescing combo :)